File: examples/custom-config/charts.m
close all; clear; clc

M = csvread('data.csv',0,1);
M = M'; % transpose so that results are column vectors
overhead = median(M(:,1:2:end))
ops = M(:,2:2:end);
pows = 0:1:20;
sizes = 2.^pows;

%% box plot
boxplot(ops-overhead)
grid on
xticklabels('2^'+string(pows))
title('Persistent Vec Apply')
xlabel('size')
ylabel('runtime (ns)')
ax = gca;
ax.FontSize = 24;

%% histogram
figure
idxs = fliplr(5+1:5:21);
for i = idxs
    histogram(ops(:,i)-overhead(i))
    hold on
end
title('Histogram of Persistent Vec Apply (sizes 32^n)')
legend('size ' + string(sizes(idxs)))
xlabel('runtime (ns)')
ylabel('occurences')
ax = gca;
ax.FontSize = 24;

File: examples/custom-config/charts.py
import matplotlib
import matplotlib.pyplot as plt
import numpy as np

matplotlib.rcParams.update({'font.size': 22})

results = np.genfromtxt("data.csv",delimiter=",")
results = results[:,1:].T # transpose so that results are column vectors
overhead = np.median(results[:,0::2],0)
ops = results[:,1::2]
pows = np.arange(0,21)
sizes = 2**pows

fig, ax = plt.subplots()
plt.boxplot(ops-overhead)
plt.grid(True)
ax.set_xticklabels(map(lambda p: "2^"+str(p), pows))
plt.title('Persistent Vec Apply')
plt.xlabel('size')
plt.ylabel('runtime (ns)')

plt.figure()
idxs = np.arange(20,4,-5)
for i in idxs:
  plt.hist(ops[:,i]-overhead[i])
plt.title('Histogram of Persistent Vec Apply (sizes 32^n)')
plt.legend(map(lambda s: "size "+str(s), sizes[idxs]))
plt.xlabel('runtime (ns)')
plt.ylabel('occurences')

plt.show()
